firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -124,6 +126,7 @@ public Thread newThread(Runnable r) {
     this.utils = utils;
     this.iidStore = iidStore;
     this.fidGenerator = fidGenerator;
+    this.networkExecutor = Executors.newSingleThreadExecutor();
```

Comment:
Q1: Why is this not injected like all the other variables and created by the other constructor method?
Q2: Is there a reason that we have different ways how backgroundExecutor and networkExecutor are initialized?

Reply:
Q2: Idea was to simplify the creation but changed it now to replicate the backgroundExecutor. Reasoning: we had a bug reported by some internal team that checks for thread tags and has whitelisted only firebase-installations-executor. 

Q1: The network executor is an impl detail specific to FIS. Adding multiple executors in the constructor could lead to the confusion. I dont see how we can benefit from initializing through injection.